pandas/_libs/lib.pyx

```diff
@@ -994,8 +994,8 @@ cdef inline bint c_is_list_like(object obj, bint allow_sets) except -1:
         isinstance(obj, abc.Iterable)
         # we do not count strings/unicode/bytes as list-like
         and not isinstance(obj, (str, bytes))
-        # exclude zero-dimensional numpy arrays, effectively scalars
-        and not (util.is_array(obj) and obj.ndim == 0)
+        # exclude zero-dimensional duck arrays, effectively scalars
+        and not (hasattr(obj, "ndim") and obj.ndim == 0)
```

Comment:
you might be able to simplify this using `getattr`:
```suggestion
        and getattr(obj, "ndim", 0) != 0
```

Reply:
Awesome thanks!